README.md

```diff
@@ -0,0 +1,140 @@
+a specific run, use `--no-history-check`.
+
+## Writing Features
+
+Developers can write workflows and activities that represent a SDK "feature". These are organized into directories under
+the `features/` directory.
+
+In addition to code for the feature, there are configuration settings that can be set in `.config.json`. The possible
+settings are:
+
+* `go`
+  * `minVersion` - Minimum version in Go this feature should be run in. The feature will be skipped in older versions.
+
+There are also files in the `history/` subdirectory which contain history files used during run. See the
+"History Checking" and "Generating History" sections for more info.
+
+### Best Practices
+
+* Try to only demonstrate/test one feature per feature directory.
+  * Code should be kept as short and clear as possible.
+  * No need to over-assert on a bunch of values, just confirm that the feature does what is expected via its output.
+* A Go feature should be in `feature.go`.
+  * For incompatible versions, different files like `feature_pre1.11.0.go` can be present using build tags
+* A Java feature in `feature.java`.
+* A TypeScript feature should be in `feature.ts` for all non-workflow code and `feature.workflow.ts` for all workflow
+  code.
+* Add a README.md to each feature directory.
+  * README should have a title summarizing the feature (only first letter needs to be in title case), then a short
+    paragraph explaining the feature and its purpose, and then optionally another paragraph explaining details of the
+    specific code steps.
+  * Other sections can also explain decisions made in language specific ways or other details about versions/approaches.
+  * Feel free to add links and more text as necessary.
+* Verification/regression feature directories for bugs should be under `features/bugs/<lang>`.
+  * Ideally the checking of the result has a version condition that shows in earlier versions it should fail and in
+    newer versions it should succeed.
+* The more languages per non-bug feature, the better. Try not to create non-bug features that use specific language
+  constructs unless that's the purpose of the feature.
+* Refactor liberally to create shortcuts and better harnesses for making features easy to read and write.
+  * This is not a library used by anyone, there are no backwards compatibility concerns. If one feature uses something
+    that has value to another, extract and put in helper/harness and have both use it.
+* History should be generated for each feature on the earliest version the feature is expected to work at without
+  history changes.
+
+#### Generating History
+
+To generate history, run the same test (see the "Running" section) for the version to generate at, but use the
+`--generate-history` option. When generating history, only one test can be specified and the version of the SDK must be
+specified.
+
+## Development
```

Comment:
I need to remove some of this stuff history file stuff. This is leftover from development.

Reply:
Removed